src/TestingDevice.java

```diff
@@ -128,11 +145,20 @@ public Device(boolean isRandom) {
      * @param size the number of bits stored in this device
      * @param bitsPerPeek the number of bits to disclose via peek or set via poke
      */
-    public Device(int size, int bitsPerPeek) {
+    public TestingDevice(int size, int bitsPerPeek) {
         this.size = size;
         boolean[] initialBits = new boolean[this.size];
+        boolean allTrue = true;
+        boolean allFalse = true;
         for (int i = 0; i < size; i++) {
-            if (Math.random() > 0.5) initialBits[i] = true;
+            if (Math.random() > 0.5) { initialBits[i] = true; allFalse = false; }
```

Comment:
This logic ensures that the last bit will be different, if all bits before it are the same. This is to help us ensure that the Device we are testing does not happen to be unlocked before we even spin it once.